Client/Frontend/Sync/BraveCore/BraveSyncQRCodeModel.swift

```diff
@@ -0,0 +1,142 @@
+        self.version = version
+        self.syncHexCode = syncHexCode
+        self.notValidAfter = Int64(notValidAfter)
+    }
+    
+    private enum CodingKeys: String, CodingKey {
+        case version
+        case syncHexCode = "sync_code_hex"
+        case notValidAfter = "not_after"
+    }
+    
+    var jsonData: Data? {
+        do {
+            return try JSONEncoder().encode(self)
+        } catch {
+            log.error("Error converting QR Code to JSON: \(error)")
+        }
+        return nil
+    }
+    
+    var jsonString: String? {
+        if let data = jsonData {
+            return String(data: data, encoding: .utf8)
+        }
+        return nil
+    }
+    
+    static func from(string: String) -> BraveSyncQRCodeModel? {
+        if let data = string.data(using: .utf8) {
+            do {
+                return try JSONDecoder().decode(Self.self, from: data)
+            } catch {
+                log.error("Error converting QR Code to BraveSyncQRCodeModel: \(error)")
+            }
+        }
+        return nil
+    }
+    
+    func validate(syncAPI: BraveSyncAPI) -> BraveSyncQRCodeError {
+        // Older verison of the QR code (not backwards compatible)
+        if version < BraveSyncQRCodeModel.currentlySupportedVersion {
+            return .insecure
+        }
+        
+        // Newer version of the QR code (not supported)
+        if version > BraveSyncQRCodeModel.currentlySupportedVersion {
+            return .newerVersion
+        }
+        
+        // Currently supported version
+        if notValidAfter < BraveSyncQRCodeModel.braveSyncEpoch {
+            // Impossible to have a QR code of this version
+            // before the brave sync epoch
+            return .invalidFormat
+        }
+        
+        // Date validation
+        let minDate = Date().timeIntervalSince1970 - BraveSyncQRCodeModel.validityDuration
+        let maxDate = Date().timeIntervalSince1970 + BraveSyncQRCodeModel.validityDuration
+        
+        // Currently Expired (QR code date is too far into the past)
+        if TimeInterval(notValidAfter) <= minDate {
+            return .expired
+        }
+        
+        // Too future (QR code date date is too far into the future)
+        if TimeInterval(notValidAfter) >= maxDate {
+            return .futureDate
+        }
+        
+        // Invalid empty sync code
+        if syncHexCode.isEmpty {
+            return .invalidFormat
+        }
+        
+        // Invalid hex sync code
+        if syncHexCode.filter(\.isHexDigit).count != syncHexCode.count {
+            return .invalidFormat
+        }
+        
+        // Invalid sync code validation
+        let codeWords = syncAPI.syncCode(fromHexSeed: syncHexCode)
+        if codeWords.isEmpty {
+            return .invalidFormat
+        }
+        
+        let codeWordsCount = codeWords.separatedBy(" ").count
+        if codeWordsCount < 24 || codeWordsCount > 25 {
```

Comment:
This should always be 24. The change for 25 words are no longer in-scope for Phase 1.

Reply:
Fixed.